src/table.rs

```diff
@@ -247,6 +258,9 @@ impl ValueTable {
 			};
 
 			if part == 0 {
+				// [Review] Skipping 6 first byte does not make much sense to me.
+				// It is used to build key material, but part of the key material
+				// is overwrite, so we don't fully compare the 32 bytes.
```

Comment:
Value table always contains 26 bytes of the key. The rest are checked in the index. The index start with 16 bits of chunk index and 38 bit of key material, so 54 key bits total. After the first reindex, chunk index increases to 17 bit and the key material decreases to 37, but total is still 54. And so on.

Reply:
:facepalm: works indeed.
I keep liking the idea to store the full hashed key, because it would allow restoring/rebuilding indexes from value table only, not that interesting though.